### PR TITLE
Preserve build id on import

### PR DESCRIPTION
**Pull request by SonOfBytes**
When performing an `convox build import` the build id of the imported package is not preserved in the target build package.

See issue https://github.com/convox/rack/issues/2025

This sets the build id based on the import file build id.

However, `make test` fails - any pointers to fix them up if this is the right direction would be appreciated.

### CODE REVIEW

**Comment by SonOfBytes** [comment]
I suspect that travis is not running an updated sync/templates.go - locally `make test` runs.

```
$ PATH=$GOPATH/bin:$PATH make test
env PROVIDER=test CONVOX_WAIT= bin/test
?   	github.com/convox/rack/api	[no test files]
?   	github.com/convox/rack/api/awsutil	[no test files]
?   	github.com/convox/rack/api/cache	[no test files]
?   	github.com/convox/rack/api/cmd/build	[no test files]
?   	github.com/convox/rack/api/cmd/build/source	[no test files]
?   	github.com/convox/rack/api/cmd/monitor	[no test files]
ok  	github.com/convox/rack/api/controllers	0.143s	coverage: 34.1% of statements
?   	github.com/convox/rack/api/crypt	[no test files]
?   	github.com/convox/rack/api/helpers	[no test files]
?   	github.com/convox/rack/api/httperr	[no test files]
ok  	github.com/convox/rack/api/models	0.094s	coverage: 11.2% of statements
?   	github.com/convox/rack/api/models/fixtures	[no test files]
?   	github.com/convox/rack/api/structs	[no test files]
?   	github.com/convox/rack/api/workers	[no test files]
?   	github.com/convox/rack/changes	[no test files]
?   	github.com/convox/rack/changes/cmd/changed	[no test files]
ok  	github.com/convox/rack/client	0.034s	coverage: 8.9% of statements
?   	github.com/convox/rack/cmd/changes	[no test files]
ok  	github.com/convox/rack/cmd/convox	27.410s	coverage: 3.5% of statements
ok  	github.com/convox/rack/cmd/convox/appinit	0.014s	coverage: 20.5% of statements
?   	github.com/convox/rack/cmd/convox/appinit/templates	[no test files]
?   	github.com/convox/rack/cmd/convox/helpers	[no test files]
ok  	github.com/convox/rack/cmd/convox/stdcli	0.031s	coverage: 42.6% of statements
?   	github.com/convox/rack/cmd/convox/templates	[no test files]
?   	github.com/convox/rack/cmd/crypt	[no test files]
ok  	github.com/convox/rack/manifest	0.042s	coverage: 49.8% of statements
?   	github.com/convox/rack/provider	[no test files]
ok  	github.com/convox/rack/provider/aws	7.434s	coverage: 53.8% of statements
?   	github.com/convox/rack/provider/aws/lambda/cfevents	[no test files]
ok  	github.com/convox/rack/provider/aws/lambda/fluentd	0.010s	coverage: 21.3% of statements
?   	github.com/convox/rack/provider/aws/lambda/formation	[no test files]
?   	github.com/convox/rack/provider/aws/lambda/formation/handler	[no test files]
?   	github.com/convox/rack/provider/aws/lambda/lifecycle	[no test files]
?   	github.com/convox/rack/sync	[no test files]
?   	github.com/convox/rack/test	[no test files]
```

**Comment by MiguelMoll** [comment]
I don't think the tests failing is due to of this change. I think its a cli change that will be fixed soon which this PR could then rebase off master.

File: provider/aws/builds.go [hunk 1]
```diff
@@ -339,6 +339,7 @@ func (p *AWSProvider) BuildImport(app string, r io.Reader) (*structs.Build, erro
 				log.Error(err)
 				return nil, err
 			}
+			targetBuild.Id = sourceBuild.Id
```
Comment (MiguelMoll) [inline provider/aws/builds.go]: I believe this line would be set around 410-411 with the other `targetBuild` properties.

Comment (SonOfBytes) [inline provider/aws/builds.go]: I thought putting it here was better as target Id is used around line 385

Comment (MiguelMoll) [inline provider/aws/builds.go]: That's a good catch. Let's leave it here. 👍

**Comment by SonOfBytes** [comment]
I have a check to ensure that the build does not already exist.  I also realised there was a race condition of build.json was not the first file in the tar.  So now tag and push after the main image build loop.

I'm unsure how to fix the import build test as it now has a dependency on BuildGet.  Guidance welcome or if this needs to go in another direction let me know.

**Comment by SonOfBytes** [comment]
I have the tests working - there may be a more proper way of achieving this :P

**Comment by MiguelMoll** [comment]
[Preserve build id on import](https://trello.com/c/R59fT5Iu/2149-preserve-build-id-on-import)

File: provider/aws/error_test.go [hunk 2]
```diff
@@ -9,3 +16,10 @@ func (e errorNotFound) Error() string {
 func (e errorNotFound) NotFound() bool {
 	return true
 }
+
+func TestNoSuchBuild_Error(t *testing.T) {
+	err := aws.NoSuchBuild("B12345")
+	if assert.Error(t, err) {
+		assert.Equal(t, errors.New("no such build: B12345").Error(), err.Error())
```
Comment (MiguelMoll) [inline provider/aws/error_test.go]: We could take advantage of `assert.EqualError` here and not wrap the string with a new error.

```
assert.EqualError(t, err,  expectedErrorString)
```

Comment (SonOfBytes) [inline provider/aws/error_test.go]: nice catch - TIL

**Comment by MiguelMoll** [comment]
Moving forward via #2091

**Pull request closed without merging**